src/test/stress/stress_test.go

```diff
@@ -0,0 +1,809 @@
+	if err != nil {
+		return err
+	}
+	err = countAndLog(receipts)
+	return err
+}
+
+func TestStressSingle(t *testing.T) {
+	skipCI(t)
+	setup, err := createSetup(true)
+	if err != nil {
+		log.Fatal("could not create setup", err)
+	}
+	env, err := createStressEnvironment(context.Background(), setup)
+	if err != nil {
+		log.Fatal("could not set up environment", err)
+	}
+	err = transact(setup, &env, 1)
+	if err != nil {
+		log.Printf("failure %s", err)
+		t.Fail()
+	}
+}
+
+func TestStressDualWait(t *testing.T) {
+	skipCI(t)
+	setup, err := createSetup(true)
+	if err != nil {
+		log.Fatal("could not create setup", err)
+	}
+	env, err := createStressEnvironment(context.Background(), setup)
+	if err != nil {
+		log.Fatal("could not set up environment", err)
+	}
+	env.WaitOnEverySubmit = true
+
+	err = transact(setup, &env, 2)
+	if err != nil {
+		log.Printf("failure %s", err)
+		t.Fail()
+	}
+}
+
+// run with `go test -test.v -run TestStressDualNoWait`
+func TestStressDualNoWait(t *testing.T) {
+	skipCI(t)
+	setup, err := createSetup(true)
+	if err != nil {
+		log.Fatal("could not create setup", err)
+	}
+	env, err := createStressEnvironment(context.Background(), setup)
+	if err != nil {
+		log.Fatal("could not set up environment", err)
+	}
+
+	err = transact(setup, &env, 2)
+	if err != nil {
+		log.Printf("failure %s", err)
+		t.Fail()
+	}
+}
+
+// send two transactions in the same block by the same sender with the same identityPrefix
+func TestStressDualDuplicatePrefix(t *testing.T) {
+	skipCI(t)
+	setup, err := createSetup(true)
+	if err != nil {
+		log.Fatal("could not create setup", err)
+	}
+	env, err := createStressEnvironment(context.Background(), setup)
+	if err != nil {
+		log.Fatal("could not set up environment", err)
+	}
+	prefix, err := createIdentity()
+	if err != nil {
+		log.Fatal("error creating prefix", err)
+	}
+	var prefixes []shcrypto.Block
+	prefixes = append(prefixes, prefix)
+	prefixes = append(prefixes, prefix)
+	env.IdentityPrefixes = prefixes
+
+	err = transact(setup, &env, 2)
+	if err != nil {
+		log.Printf("failure %s", err)
+		t.Fail()
+	}
+}
+
+// send many transactions as quickly as possible.
+func TestStressManyNoWait(t *testing.T) {
+	skipCI(t)
+	setup, err := createSetup(true)
+	if err != nil {
+		log.Fatal("could not create setup", err)
+	}
+	env, err := createStressEnvironment(context.Background(), setup)
+	if err != nil {
+		log.Fatal("could not set up environment", err)
+	}
+
+	err = transact(setup, &env, 47)
+	if err != nil {
+		log.Printf("failure %s", err)
+		t.Fail()
+	}
+}
+
+func TestStressExceedEncryptedGasLimit(t *testing.T) {
+	skipCI(t)
+	setup, err := createSetup(true)
+	if err != nil {
+		log.Fatal("could not create setup", err)
+	}
+	env, err := createStressEnvironment(context.Background(), setup)
+	if err != nil {
+		log.Fatal("could not set up environment", err)
+	}
+
+	env.TransactGasLimitFn = func(value *big.Int, data []byte, toAddress *common.Address, i, count int) uint64 {
```

Comment:
Is it intended that `value` iosnt used and the limit is hardcoded?

Reply:
When I designed the interface
https://github.com/shutter-network/encrypting-rpc-server/pull/13/files#diff-2fde392c0c13754486fec812f19b966baf42611de448feb45f4d1348c1d8b03fR62
I imagined test cases, where we want to variate gas limit depending on the `value`. But it is probably YAGNI. I will remove the argument.